### DISCUSSION TITLE
Github Copilot error code: 'ECONNRESET',   errno: -4077. in Visual studio code when working in the office network.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#110919] [Copilot Conversations]
### Select Topic Area

Question

### Body

Github copilot when installed in Visual Studio Code and connected to the Office network gives a connection error with error code: 'ECONNRESET',   errno: -4077.
Please suggest how do I solve this. Is there any specific websites need to be white-listed by the Organizational Network team for this Github copilot extension in VSC to work?

Any help is appreciated.
Thanks 
Anil

**Comment by Akash1134** [comment]
Hi @tripathyanil ,

As we [announced in September](https://github.blog/changelog/2023-09-25-breaking-change-to-github-copilot-service-endpoints/), we updated the API service endpoints used by Copilot Chat. On February 1, 2024, we will deprecate the Copilot Chat API endpoints currently being routed through https://copilot-proxy.githubusercontent.com. Instead, these requests will go through https://api.githubcopilot.com. Versions 0.8.0 and later of the Copilot Chat extension for VS Code and versions 0.1.1817.27579 and later of the Copilot Chat extension for Visual Studio already route chat traffic through https://api.githubcopilot.com. All versions of the Copilot plugin for JetBrains IDEs already route chat traffic through https://api.githubcopilot.com.

To ensure Copilot Chat continues working from February 1, 2024, please update to the most recent version of the Copilot extension and ensure your firewall and network settings allow communication to https://api.githubcopilot.com.

Please refer [Important Updates: Copilot Chat API endpoints and Copilot Content Exclusions](https://github.com/orgs/community/discussions/101438)

Thanks!

**Reply by valenradovich** [reply]
hey man, i hope your doing well!

i'm having this same issue with github copilot when my pc is connected to my company VPN. But is weird because Copilot Chat is working, the one who's not working is the regular Copilot and the autocomplete. My error message is the same: 'FetchError', 'ECONNRESET', -4077, 'read'.

i hope you can help me, ty!

**Reply by audreyeternal** [reply]
I had the same problem.

**Reply by smarudhi** [reply]
I'm also facing same issue and copilot chat isn't working as sign in is failed for chat but GitHub sign in is successful.

**Reply by valenradovich** [reply]
in my case the firewall was blocking the request

**Comment by Akash1134** [comment]
Hi All, 


Thank you to everyone who has participated in this discussion about the 'ECONNRESET' error when using GitHub Copilot in corporate networks. After reviewing the reports and solutions shared, we'd like to provide an official consolidated response.

## Root Cause

This error typically occurs when corporate firewalls or network security policies block GitHub Copilot's required endpoints. As noted in our [September 2023 announcement](https://github.blog/changelog/2023-09-25-breaking-change-to-github-copilot-service-endpoints/), GitHub Copilot has updated its API service endpoints.

## Required Endpoints

To ensure GitHub Copilot functions properly in your corporate environment, please ensure the following domains are allowlisted in your network:

- `https://api.githubcopilot.com`
- `https://copilot-proxy.githubusercontent.com` (legacy endpoint, but may still be used by older versions)
- `https://default.exp-tas.com` (for telemetry)

## Solutions

Please start with the [documentation](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot) to troubleshoot network related errors. Next Steps: 

1. **Update Your Copilot Extensions**
   Ensure you're using the latest version of GitHub Copilot and Copilot Chat extensions, as newer versions implement the current API endpoints.

2. **Network Configuration**
   Work with your IT department to allowlist the domains listed above in your corporate firewall/proxy.
   
3. **Proxy Settings**
   If your organization uses a proxy server, configure VS Code to use it:
   - Go to Settings > Search for "proxy"
   - Configure the HTTP proxy settings appropriately

4. **VPN Issues**
   Some users have reported that split-tunneling VPN configurations can help by routing GitHub Copilot traffic directly rather than through the corporate network.

5. **Temporary Workaround**
   As a last resort, some users have found success by temporarily disconnecting from their corporate network or VPN when using Copilot, though this may not be feasible in all work environments.

## Next Steps

If you continue experiencing issues after implementing these solutions:
- Check if Copilot Chat works while regular Copilot doesn't (as reported by some users)
- Include specific error messages and VS Code/Copilot version information when seeking further support
- Consider filing a detailed support request through [GitHub Support](https://support.github.com)

This discussion will now be marked as resolved. If you encounter different issues with GitHub Copilot, please start a new discussion with specific details.

Thank you for your participation and patience!
